.github/workflows/release.yml

```diff
@@ -0,0 +1,48 @@
+name: Release
+
+on:
+  push:
+    branches:
+      - main
+      - beta
+
+jobs:
+  release:
+    name: Release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out Git repository
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+          persist-credentials: false
+          token: ${{ secrets.GH_TOKEN }}
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: 16
+
+      - name: Get yarn cache
+        uses: actions/cache@v3
+        id: yarn-cache
+        with:
+          path: |
+            **/node_modules
+            ~/.cache
+          key: ${{ runner.os }}-node-16-yarn-${{ hashFiles('**/yarn.lock') }}-${{ hashFiles('.github/workflows/**.yml') }}
+
+      - name: Install Node.js dependencies
+        if: steps.yarn-cache.outputs.cache-hit != 'true'
+        run: yarn install --frozen-lockfile
+
+      - run: yarn lint
+
+      - run: yarn test:unit
+      - run: (yarn server &) && sleep 1 && yarn test:integration
+
+      - run: yarn build
```

Comment:
Could we also run everything up to this point on every PR? To avoid unexpected issues when merging.

Reply:
You can create a new `pr.yml` with the same steps but following `on`:

```
name: Pull Request

on:
  pull_request:
    branches:
      - main
      - beta
```

Reply:
Should be better now! I added `if: ${{ contains(github.ref, 'refs/heads/main') || contains(github.ref, 'refs/heads/beta') }}`

Reply:
Interesting approach. In other projects we have a separate action, but I kind of like this approach - do not repeat yourself! It might be a bit harder to read, but I like it 👍